skimage/exposure/hdr.py

```diff
@@ -0,0 +1,311 @@
+import numpy as np
+
+
+def get_radiance(images, exposure, radiance_map):
+    """
+    Return the radiance image for a series of images based upon a camera
+    response function.
+
+    Parameters
+    ----------
+    images: list
+        List of images in the for of numpy arrays. Either mono or color
+        (RGB, MxNx3).
+    exposure : numpy 1D array
+        Array of exposure times in seconds.
+    radiance_map : numpy array
+        Array mapping the counts to radiance
+
+    Returns
+    -------
+    hdr : numpy array
+        Resulting image with radiance values.
+    """
+
+    den = np.ones(images[0].shape)
+    num = np.zeros(images[0].shape)
+    wf = np.vectorize(_weight_func)
+    for idx, im in enumerate(images):
+        gij = im.copy()
+        # For colour
+        if im.ndim == 3:
+            for ii in range(im.shape[2]):
+                gij[:, :, ii] = radiance_map[im[:, :, ii] + 1, ii]
+        else:
+            gij[:, :] = radiance_map[im[:, :, ii] + 1]
+        gij = gij - np.log(exposure[idx])
+
+        wij = wf(gij)
+        num = num + wij * gij
+        den = den + wij
+
+    return num / den
+
+
+def make_hdr(images, exposure, radiance_map, depth=16):
+    """
+    Compute the HDR image from a series of images with a given radiance
+    mapping.
+
+    Parameters
+    ----------
+    images: list or ImageCollection
+        List of images as numpy arrays. Either grayscale or color
+        (RGB).
+        Can also be an ImageCollection
+    exposure : numpy 1D array
+        Array of exposure times in seconds.
+        Images some times have these in the exif information.
+    radiance_map : numpy array
+        Array (idx) mapping counts to radiance value, if input is RGB this must
+        be Nx3. See `get_crf`.
+    depth : int, optional
+        Pixel depth.
+
+    Returns
+    -------
+    hdr : numpy array
+        The HDR image, either grayscale or RGB depending on input, in ln(E).
+
+    References
+    ----------
+    .. [1]  Debevec, P. E., & Malik, J. (1997). SIGGRAPH 97 Conf. Proc.,
+            August, 3-8. DOI:10.1145/258734.258884
+    """
+    B = np.log(np.array(exposure))
+
+    if images[0].ndim == 3:
```

Comment:
Multi-channel images are now managed using the `channel_axis` parameter (please see #4294)

Reply:
Just to make sure that I understand, this needs to check which dimension is the RGB dimension via the `channel_axis` parameter?

Reply:
Yes, that's it :wink:

Reply:
And what is supposed to happen if it doesn't exist (i.e. a list of numpy arrays)?

Reply:
if `channel_axis` is `None`, the image is assumed to be grayscale.

Reply:
I understand that `None` would mean greyscale. 
Though doing for instance 
`if images[0].channel_axis` 
would run into a AttributeError for pure NumPy arrays, which from my understanding of scikit-image methods should support.
As I understand from the coordinate conventions for scikit-image, the last index should be colour. Is the standard to not trust this and then having to find it manually?

Reply:
No, sorry, I meant that we must add a `channel_axis` argument to the function signature.
Instead of assuming that the image is multichannel if `image.ndim = 3` and that the channels are in the last axis, we use the `channel_axis` argument that is specifies on which axis are the channels.

Reply:
Now I'm following. I guess the parameter defaults to `None`?

Reply:
Yes, exactly :wink:

Reply:
I have now tried to implement this.